Modules/OptimismMainModule.php

```diff
@@ -30,7 +30,8 @@ function initialize()
-        $this->handles_regex = '/(.*)\.box/';
+        // This is the full list of names supported by 3DNS, including .box
+        $this->handles_regex = '/(.*)\.(box|ac|academy|accountant|accountants|actor|agency|ai|airforce|apartments|app|archi|army|art|asia|associates|at|attorney|auction|audio|auto|autos|baby|band|bar|bargains|bayern|beauty|beer|best|bible|bid|bike|bingo|bio|biz|black|blackfriday|blog|blue|boats|bond|boston|boutique|builders|business|buzz|ca|cab|cafe|cam|camera|camp|capital|car|cards|care|careers|cars|casa|cash|casino|catering|cc|center|cfd|charity|chat|cheap|christmas|church|city|claims|cleaning|click|clinic|clothing|cloud|club|co|coach|codes|coffee|college|com|community|company|computer|condos|construction|consulting|contact|contractors|cooking|cool|coupons|courses|credit|creditcard|cricket|cruises|cyou|dance|date|dating|de|dealer|deals|degree|delivery|democrat|dental|dentist|design|dev|diamonds|diet|digital|direct|directory|discount|doctor|dog|domains|download|earth|education|email|energy|engineer|engineering|enterprises|equipment|estate|eu|events|exchange|expert|exposed|express|fail|faith|family|fans|farm|fashion|feedback|film|finance|financial|fish|fishing|fit|fitness|flights|florist|flowers|football|forsale|foundation|fun|fund|furniture|futbol|fyi|gallery|game|games|garden|gay|gift|gifts|gives|glass|global|gmbh|gold|golf|graphics|gratis|green|gripe|group|guide|guitars|guru|hair|haus|health|healthcare|help|hiphop|hiv|hockey|holdings|holiday|homes|horse|hospital|host|hosting|house|icu|immo|immobilien|in|inc|industries|info|ink|institute|insure|international|investments|io|irish|ist|istanbul|jetzt|jewelry|juegos|kaufen|kids|kim|kitchen|la|land|lat|law|lawyer|lease|legal|lgbt|life|lighting|limited|limo|link|live|llc|loan|loans|lol|london|love|ltd|luxe|maison|makeup|management|market|marketing|mba|me|media|melbourne|memorial|men|menu|miami|mobi|moda|moe|mom|money|monster|mortgage|motorcycles|movie|mx|navy|net|network|news|ninja|nl|nyc|observer|one|online|ooo|org|organic|page|partners|parts|party|pet|ph|photo|photography|photos|pics|pictures|pink|pizza|place|plumbing|plus|poker|press|pro|productions|promo|properties|property|protection|pub|pw|quest|racing|realty|recipes|red|rehab|reise|reisen|rent|rentals|repair|report|republican|rest|restaurant|review|reviews|rip|rocks|rodeo|run|sale|salon|sarl|sbs|school|schule|science|security|services|sexy|sh|shiksha|shoes|shop|shopping|show|singles|site|ski|skin|soccer|social|software|solar|solutions|space|storage|store|stream|studio|study|style|sucks|supplies|supply|support|surf|surgery|sydney|systems|tattoo|tax|taxi|team|tech|technology|tel|tennis|theater|theatre|tickets|tienda|tips|tires|today|tools|top|tours|town|toys|trade|training|travel|tube|tv|uk|university|uno|us|vacations|vegas|ventures|vet|viajes|video|villas|vin|vip|vision|vodka|vote|voyage|watch|webcam|website|wedding|whoswho|wiki|win|wine|work|works|world|ws|wtf|xyz|yachts|yoga|zone)/';
```

Comment:
I believe this should be put in a separate variable at least.
What do you think if we replace it this way:
```php
$this->handle_roots = array_unique(['box', 'ac', 'academy', 'accountant', 'accountants', 'actor', 'agency', 'ai', 'airforce', 'apartments', 'app', 'archi', 'army', 'art', 'asia', 'associates', 'at', 'attorney', 'auction', 'audio', 'auto', 'autos', 'baby', 'band', 'bar', 'bargains', 'bayern', 'beauty', 'beer', 'best', 'bible', 'bid', 'bike', 'bingo', 'bio', 'biz', 'black', 'blackfriday', 'blog', 'blue', 'boats', 'bond', 'boston', 'boutique', 'builders', 'business', 'buzz', 'ca', 'cab', 'cafe', 'cam', 'camera', 'camp', 'capital', 'car', 'cards', 'care', 'careers', 'cars', 'casa', 'cash', 'casino', 'catering', 'cc', 'center', 'cfd', 'charity', 'chat', 'cheap', 'christmas', 'church', 'city', 'claims', 'cleaning', 'click', 'clinic', 'clothing', 'cloud', 'club', 'co', 'coach', 'codes', 'coffee', 'college', 'com', 'community', 'company', 'computer', 'condos', 'construction', 'consulting', 'contact', 'contractors', 'cooking', 'cool', 'coupons', 'courses', 'credit', 'creditcard', 'cricket', 'cruises', 'cyou', 'dance', 'date', 'dating', 'de', 'dealer', 'deals', 'degree', 'delivery', 'democrat', 'dental', 'dentist', 'design', 'dev', 'diamonds', 'diet', 'digital', 'direct', 'directory', 'discount', 'doctor', 'dog', 'domains', 'download', 'earth', 'education', 'email', 'energy', 'engineer', 'engineering', 'enterprises', 'equipment', 'estate', 'eu', 'events', 'exchange', 'expert', 'exposed', 'express', 'fail', 'faith', 'family', 'fans', 'farm', 'fashion', 'feedback', 'film', 'finance', 'financial', 'fish', 'fishing', 'fit', 'fitness', 'flights', 'florist', 'flowers', 'football', 'forsale', 'foundation', 'fun', 'fund', 'furniture', 'futbol', 'fyi', 'gallery', 'game', 'games', 'garden', 'gay', 'gift', 'gifts', 'gives', 'glass', 'global', 'gmbh', 'gold', 'golf', 'graphics', 'gratis', 'green', 'gripe', 'group', 'guide', 'guitars', 'guru', 'hair', 'haus', 'health', 'healthcare', 'help', 'hiphop', 'hiv', 'hockey', 'holdings', 'holiday', 'homes', 'horse', 'hospital', 'host', 'hosting', 'house', 'icu', 'immo', 'immobilien', 'in', 'inc', 'industries', 'info', 'ink', 'institute', 'insure', 'international', 'investments', 'io', 'irish', 'ist', 'istanbul', 'jetzt', 'jewelry', 'juegos', 'kaufen', 'kids', 'kim', 'kitchen', 'la', 'land', 'lat', 'law', 'lawyer', 'lease', 'legal', 'lgbt', 'life', 'lighting', 'limited', 'limo', 'link', 'live', 'llc', 'loan', 'loans', 'lol', 'london', 'love', 'ltd', 'luxe', 'maison', 'makeup', 'management', 'market', 'marketing', 'mba', 'me', 'media', 'melbourne', 'memorial', 'men', 'menu', 'miami', 'mobi', 'moda', 'moe', 'mom', 'money', 'monster', 'mortgage', 'motorcycles', 'movie', 'mx', 'navy', 'net', 'network', 'news', 'ninja', 'nl', 'nyc', 'observer', 'one', 'online', 'ooo', 'org', 'organic', 'page', 'partners', 'parts', 'party', 'pet', 'ph', 'photo', 'photography', 'photos', 'pics', 'pictures', 'pink', 'pizza', 'place', 'plumbing', 'plus', 'poker', 'press', 'pro', 'productions', 'promo', 'properties', 'property', 'protection', 'pub', 'pw', 'quest', 'racing', 'realty', 'recipes', 'red', 'rehab', 'reise', 'reisen', 'rent', 'rentals', 'repair', 'report', 'republican', 'rest', 'restaurant', 'review', 'reviews', 'rip', 'rocks', 'rodeo', 'run', 'sale', 'salon', 'sarl', 'sbs', 'school', 'schule', 'science', 'security', 'services', 'sexy', 'sh', 'shiksha', 'shoes', 'shop', 'shopping', 'show', 'singles', 'site', 'ski', 'skin', 'soccer', 'social', 'software', 'solar', 'solutions', 'space', 'storage', 'store', 'stream', 'studio', 'study', 'style', 'sucks', 'supplies', 'supply', 'support', 'surf', 'surgery', 'sydney', 'systems', 'tattoo', 'tax', 'taxi', 'team', 'tech', 'technology', 'tel', 'tennis', 'theater', 'theatre', 'tickets', 'tienda', 'tips', 'tires', 'today', 'tools', 'top', 'tours', 'town', 'toys', 'trade', 'training', 'travel', 'tube', 'tv', 'uk', 'university', 'uno', 'us', 'vacations', 'vegas', 'ventures', 'vet', 'viajes', 'video', 'villas', 'vin', 'vip', 'vision', 'vodka', 'vote', 'voyage', 'watch', 'webcam', 'website', 'wedding', 'whoswho', 'wiki', 'win', 'wine', 'work', 'works', 'world', 'ws', 'wtf', 'xyz', 'yachts', 'yoga', 'zone']);
$this->handles_regex = '/(.*)\.' . implode('|', $this->handle_roots) .'');
```

My point is that it's hard to read this piece of code, and extremely easy to put duplicate value.

Reply:
Dynamic properties are deprecated in PHP, so `$this->handle_roots = ...` would return an error. This is also a slower way. Right now, the list is sorted alphabetically, except for `.box`.